### PR TITLE
Fix cross compilation

### PR DESCRIPTION
**Pull request by jdblischak**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

---

* Unable to load conda binary on Apple Silicon (reported by @nickvigilante)
* Fixes cross-compilation by adding `cross-r-base {{ r_base }}` to build requirements (follow-up to #9)
* Also updated recipe and build scripts with [bgruening/conda_r_skeleton_helper](https://github.com/bgruening/conda_r_skeleton_helper)
* Added myself as co-maintainer

### CODE REVIEW

**Review by ihnorton — APPROVED** [review]
LGTM (modulo CI - looks like a missing dep?)

**Comment by jdblischak** [comment]
@ihnorton CI is green now. I won't have write-access until after this PR is merged. This PR touches a lot of files, so please do not squash the commits

**Pull request merged**
